packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -1141,17 +1144,40 @@ export default class SchedulerTask {
                 payload.metricQuery.additionalMetrics,
                 payload.metricQuery.tableCalculations,
             );
-            await this.googleDriveClient.appendToSheet(
-                refreshToken,
-                spreadsheetId,
-                rows,
-                itemMap,
-                payload.showTableNames,
-                undefined, // tabName
-                payload.columnOrder,
-                payload.customLabels,
-                payload.hiddenFields,
-            );
+            if (payload.pivotConfig) {
+                // PivotQueryResults expects a formatted ResultRow[] type, so we need to convert it first
+                // TODO: refactor pivotQueryResults to accept a Record<string, any>[] simple row type for performance
+                const formattedRows = formatRows(rows, itemMap);
+
+                const pivotedResults = pivotResultsAsCsv(
+                    payload.pivotConfig,
+                    formattedRows,
+                    itemMap,
+                    payload.metricQuery,
+                    payload.customLabels,
+                    true, // onlyRaw
```

Comment:
I feel like `pivotResultsAsCsv` should receive an object to make it easy to read what arguments represent

Reply:
done